### PR TITLE
refactor(core): inquire when required env not found

### PR DESCRIPTION
**Pull request by gao-sun**
## Summary

inquire to generate or input when required env variable not found

## Testing

UT + arbitrary required env missing

### CODE REVIEW

File: packages/core/src/env-set/create-pool-by-env.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    const databaseDsn = assertEnv(key);
+
+    return createPool(databaseDsn, { interceptors });
+  } catch (error: unknown) {
+    const answer = await inquirer.prompt({
+      name: 'dsn',
+      message: `No Postgres DSN (${key}) found in env variables. Please input the DSN which points to Logto database:`,
+    });
+
+    if (!answer.dsn) {
```
Comment (simeng-li) [inline packages/core/src/env-set/create-pool-by-env.ts:27]: nit: do we need to validate the format of input dsn here

**Review by simeng-li — APPROVED** [review]
LGTM

File: packages/core/src/env-set/create-pool-by-env.ts (hunk 1 above)
Comment (IceHe) [reply packages/core/src/env-set/create-pool-by-env.ts:27]: Better trim and validate.
e.g. blank characters ` \t\r'` will not throw an error here.

File: packages/core/src/env-set/append-dot-env.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { appendFileSync } from 'fs';
+
+const appendDotEnv = (key: string, value: string) => {
```
Comment (wangsijie) [inline packages/core/src/env-set/append-dot-env.ts:3]: In some runtime environments like Docker, `ENV` is not passed by `.env` file. Or, the user may set `ENV` directly by OS's `export`, can we detect that and throw an error while tring to append dot env?

Comment (gao-sun) [reply packages/core/src/env-set/append-dot-env.ts:3]: per discussion add `--silent` option to directly throw error when required env not found, LOG-2254

**Pull request merged**